### PR TITLE
Return full results, and tests

### PR DESCRIPTION
**Pull request by ocke**
:v:

/cc @zendesk/vegemite

### Description
Added tests and tests and tests.
Reverted 1 breaking change: always return full result

### References
* JIRA: 

### Risks

### CODE REVIEW

File: spec/client_spec.js [hunk 1]
```diff
@@ -333,12 +331,30 @@ describe('Client', function() {
...
+          origin: subject._origin,
+          source: subject._source,
+          data: { id: requestsCount, result: {
+            'ticket.subject': 'test',
+            'ticket.requester': 'test'
+          }}
+        });
+      });
+
+      it('succeeds with errors when bulk requesting', function(done) {
```
Comment (sandlerr) [inline spec/client_spec.js]: maybe `resolves` instead of `succeeds`?

File: spec/client_spec.js [hunk 2]
```diff
@@ -357,73 +373,102 @@ describe('Client', function() {
...
-      it('does not throw an error when bulk requesting', function(done) {
-        var promise = subject.get(['ticket.subj']);
+      it('accepts an object with multiple paths', function(done) {
+        promise = subject.set({
+          'ticket.subject': 'value',
+          'ticket.description': 'value'
+        });
 
-        expect(promise).to.become({ errors: { 'ticket.subj': { message: 'No such Api' } } }).and.notify(done);
+        expect(promise).to.be.a.promise;
```
Comment (sandlerr) [inline spec/client_spec.js]: isn't this implied by the next line? or is the error message really bad when it fails?

File: spec/client_spec.js [hunk 3]
```diff
@@ -435,19 +480,48 @@ describe('Client', function() {
     });
 
     describe('#invoke', function() {
-      it('returns a promise', function() {
-        promise = subject.invoke('iframe.resize');
+      it('returns a promise', function(done) {
```
Comment (sandlerr) [inline spec/client_spec.js]: this description doesn't match what the test actually does

File: spec/client_spec.js [hunk 4]
```diff
@@ -435,19 +480,48 @@ describe('Client', function() {
...
+
+        window.addEventListener.callArgWith(1, {
+          origin: subject._origin,
+          source: subject._source,
+          data: { id: requestsCount, result: { errors: {}, 'ticket.appendText': true } }
+        });
       });
 
-      it('throw when called with an object', function() {
+      it('throws an error when the handler throws it', function(done) {
```
Comment (sandlerr) [inline spec/client_spec.js]: the handler isn't exactly throwing an error, it's just returning one

File: spec/client_spec.js [hunk 5]
```diff
@@ -435,19 +480,48 @@ describe('Client', function() {
...
+          data: { id: requestsCount, result: { errors: { 'ticket.foo': { message: 'ticket.foo unavailable' } } } }
+        });
+      });
+
+      it('throws an error when invoked with an object', function() {
         expect(function() {
           subject.invoke({
             'iframe.resize': [1]
           });
         }).to.throw(Error);
```
Comment (sandlerr) [inline spec/client_spec.js]: while you're here, can we make this assertion more specific? I'd like to make sure that this error has a useful message for developers.

**Comment by sandlerr** [comment]
I don't really like putting `expect(promise).to.be.a.promise;` everywhere. Seems like something that could maybe be provided by `chai-as-promised`?

File: spec/client_spec.js (hunk 5 above)
Comment (ocke) [inline spec/client_spec.js]: It has a good messge, it just that this test stubs it all. in reality the message would be the error message from console.

**Comment by ocke** [comment]
You're right, I removed them

File: spec/client_spec.js (hunk 4 above)
Comment (sandlerr) [inline spec/client_spec.js]: nvm, I am wrong

**Comment by sandlerr** [comment]
:+1: on green

**Comment by danielbreves** [comment]
👍

**Pull request merged**